runtime/jcl/common/jclcinit.c

```diff
@@ -280,6 +280,7 @@ jint initializeKnownClasses(J9JavaVM* vm, U_32 runtimeFlags)
 					Trc_JCL_initializeKnownClasses_ClassRefNotResolvedForInstanceFieldRef(vm->mainThread, romFieldConstantPool[i].classRefCPIndex, i);
 				}  else {
 					Trc_JCL_initializeKnownClasses_ExitError(vm->mainThread, i);
+					Trc_JCL_Assert_InitializeKnownClassesFailure();
```

Comment:
I assume it's easy to get the index from the tracepoint? It could also be added to the assert so the index is in the assert message. i.e. we could assert `(((U_32)-1) == i)`

Reply:
I don't think that will yield the value of `i` in the trace message.

Reply:
Right, trace assert only takes a boolean input, and doesn't record the actual value of `i`. That's why I kept the trace point before it which can be enabled afterwards to find out the `i`.

Reply:
Maybe we should just add a print instead of asserting?

Reply:
This code path returns `JNI_ERR` and VM will exit. A print output might be buried within console output, assertion provides a clear failure location & message.